MetaMask/metamask-extension
# fix: Redesign Signature Message date values

## **Description**

Fix date values:
- Formerly, conversations were converting values as if they were milliseconds. However, these values come from Solidity where these timestamps are unix timestamps in seconds. 
- Support -1 (no expiration) value 
- Display "expiry" as a date instead of a unix timestamp

This PR fixes the conversion.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27249?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27137

## **Manual testing steps**

1. Go to test-dapp
2. Request Permit or another TypedSign request
3. Observe deadlines and dates

## **Screenshots/Recordings**



### **Before**

#### Test-dapp Permit
<img width="320" src="https://github.com/user-attachments/assets/855fc4a5-5a1b-4575-bfda-40b5b6688fe3"> 

#### Deadline: -1
<img width="320" src="https://github.com/user-attachments/assets/0bc90b15-daa7-4537-9cd1-8727000899b2">

### **After**

#### Test-dapp Permit
<img width="320" src="https://github.com/user-attachments/assets/e1beb4f9-2b41-4ef9-8190-8264d88c9ab0"> 

#### Deadline: -1
<img width="320" src="https://github.com/user-attachments/assets/edb4d259-29a3-4558-8858-e5b619c0ce18">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
